code/game/objects/effects/portals.dm

```diff
@@ -45,20 +46,23 @@
 	return ..()
 
 /obj/effect/portal/attackby(obj/item/W, mob/user, params)
-	if(user && Adjacent(user))
-		user.forceMove(get_turf(src))
-		return TRUE
+	if(user)
+		if(get_turf(user) == get_turf(src))
+			teleport(user)
+			return TRUE
+		if(Adjacent(user))
+			user.Move(get_turf(src))
+			return TRUE
 
 /obj/effect/portal/attack_tk(mob/user)
 	return
 
-/obj/effect/portal/proc/on_entered(datum/source, atom/movable/arrived, direction)
-	SIGNAL_HANDLER
-	if(isobserver(arrived))
-		return
-	if(linked && (get_step(source, REVERSE_DIR(direction)) == get_turf(linked)))
-		return
-	teleport(arrived)
+/obj/effect/portal/Cross(atom/movable/crosser)
```

Comment:
Cross() is a pure proc thats called before the mover has even finished moving, dont use it for this.

Reply:
Roger, changing to Bumbed anyway I think since closed turfs actually block this.